Fix brew man discovery: binary fallback, base name matching, --find/--list verbs, binary alias search, section parsing, --interactive for --find, autogenerated-files workflow (#25)
toobuntu/homebrew-cask-tools: README.md

     on each app bundle.
   - Prints a security warning reminding you to verify the software.

#### Security notice

Using `--no-quarantine` bypasses macOS Gatekeeper for the installed app.
Only use this flag with software you trust.

---

### `brew purge-quarantine`

Disables macOS's Gatekeeper for the named casks by removing the
`com.apple.quarantine` and `com.apple.provenance` extended attributes from
their installed `.app` and plugin bundles (for example: `.component`, `.colorpicker`, `.saver`, `.vst3`).

#### Usage

```sh
brew purge-quarantine <cask> [<cask> ...]
```

#### Security notice

Removing quarantine bypasses macOS's Gatekeeper for the affected apps.
Please use this command only with software you trust.

---

### `brew man`

> Homebrew kegs (especially keg-only formulae like `openssl@3` or `libressl`) are

Reviewer: libressl is keg-only, but openssl is not.
```console
brew info --installed --json=v1 | jq '.[]
  | select(.name | startswith("libressl") or startswith("openssl"))
  | {name, keg_only, keg_only_reason, linked_keg}'
{
  "name": "libressl",
  "keg_only": true,
  "keg_only_reason": {
    "reason": "it conflicts with OpenSSL",
    "explanation": ""
  },
  "linked_keg": null
}
{
  "name": "openssl@3",
  "keg_only": false,
  "keg_only_reason": null,
  "linked_keg": "3.6.2"
}
```

#### Usage

```sh
brew purge-quarantine <cask> [<cask> ...]
```

#### Security notice

Removing quarantine bypasses macOS's Gatekeeper for the affected apps.
Please use this command only with software you trust.

---

### `brew man`

Homebrew kegs (especially keg-only formulae like `libressl`) are
not on the default `MANPATH`, so `man` does not reliably find their pages. When
multiple providers ship the same page name, `man` silently returns the first
match. `brew man` resolves man pages **by formula** and makes ambiguity explicit.

#### Usage

```
brew man [<section>] <formula> [<manpage>]
brew man --find [--interactive] <manpage>
brew man --list [--interactive] <formula>
```

| Use case | Syntax |
|---|---|
| View a formula's default page | `brew man openssl@3` |
| View a specific page within a formula | `brew man openssl@3 openssl.1ssl` |
| Restrict to a man section | `brew man 1 libressl openssl` |
| Find all providers of a page | `brew man --find openssl` |
| Pick from providers interactively | `brew man --find --interactive openssl` |
| List all pages a formula provides | `brew man --list libressl` |
| Pick from all formula pages | `brew man --list --interactive libressl` |

> | Render as HTML in a browser | `brew man --html curl` |

Reviewer: ```suggestion
| Render as HTML in a browser | `brew man --html curl` |
| Find all providers of a page | `brew man --find openssl` |
| Pick from providers interactively | `brew man --find --interactive openssl` |
| List all pages a formula provides | `brew man --list libressl` |
| Pick from all formula pages | `brew man --list --interactive libressl` |
```

> 

Reviewer: ```suggestion

> Results from `--find` and `--list` are shown as `provider: page` pairs.
```

> `--interactive` replaces printed results with a numbered prompt; select an entry to open it.

Reviewer: ```suggestion
`--interactive` replaces printed results with a numbered prompt; select an entry to open its page.
```

- `brew man --find --interactive openssl` — choose which provider's `openssl(1)` to view
- `brew man --list --interactive libressl` — choose which of libressl's pages to open

When no `<manpage>` is given, `brew man <formula>` defaults to the formula name.
If no man page matches the formula name, the formula's executables are tried as
fallback — for example, `brew man libressl` resolves to `openssl(1)` because
`libressl` ships a `bin/openssl` executable.

#### Arguments and flags

| Argument / Flag | Description |
|---|---|
| `[<section>]` | Optional man section number (e.g. `1`, `3`) before the formula name |
| `<formula>` | The installed formula whose keg to search (default mode) |
| `[<manpage>]` | Man page name to look up (defaults to `<formula>`) |

> | `--html`, `-H` | Render the man page as HTML and open it in a browser (respects `HOMEBREW_BROWSER` or `BROWSER`) |

Reviewer: The mutual exclusivity isn't --html vs --find/--list — it's --html vs --find/--list without --interactive. The --html flag modifies the open step, which only exists when --interactive is present. So:

--find --html → nonsensical, no open step
--find --interactive --html → coherent, open the selected page as HTML

That changes the --html description to something like:

Open the page as HTML in a browser; incompatible with --find or --list unless --interactive is also given

This will require changes to cmd/man.rb.

```

#### Security notice

Removing quarantine bypasses macOS's Gatekeeper for the affected apps.
Please use this command only with software you trust.

---

### `brew man`

Homebrew kegs (especially keg-only formulae like `libressl`) are
not on the default `MANPATH`, so `man` does not reliably find their pages. When
multiple providers ship the same page name, `man` silently returns the first
match. `brew man` resolves man pages **by formula** and makes ambiguity explicit.

#### Usage

```
brew man [<section>] <formula> [<manpage>]
brew man --find [--interactive] <manpage>
brew man --list [--interactive] <formula>
```

| Use case | Syntax |
|---|---|
| View a formula's default page | `brew man openssl@3` |
| View a specific page within a formula | `brew man openssl@3 openssl.1ssl` |
| Restrict to a man section | `brew man 1 libressl openssl` |
| Render as HTML in a browser | `brew man --html curl` |
| Find all providers of a page | `brew man --find openssl` |
| Pick from providers interactively | `brew man --find --interactive openssl` |
| List all pages a formula provides | `brew man --list libressl` |
| Pick from all formula pages | `brew man --list --interactive libressl` |

> > Results from `--find` and `--list` are shown as `provider: page` pairs.

Reviewer: Or perhaps:

```console
> `--find` results are shown as `provider: /path/to/manfile`; `--list` results are shown as `page: /path/to/manfile`.
```

Each format surfaces exactly the thing that varies, with the constant (what you searched for) implicit in the command itself.

The path is load-bearing. Two system: entries for bash are indistinguishable without it. And the openssl case shows the extension varies between providers (.1 vs .1ssl), which is meaningful information a bare page name would lose.

```console
# From an outdated version of `brew man`, for illustrative purposes only and NOT reflective of the current code:
brew man --list bash
==> bash found in:
  bash: /opt/homebrew/opt/bash/share/man/man1/bash.1
  system: /usr/share/man/man1/bash.1
  system: /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/share/man/man1/bash.1

# it seems there is no man page for `system` openssl
brew man --list openssl
==> openssl found in:
  libressl: /opt/homebrew/opt/libressl/share/man/man1/openssl.1
  openssl: /opt/homebrew/opt/openssl/share/man/man1/openssl.1ssl

brew man --list git
==> git found in:
  git: /opt/homebrew/opt/git/share/man/man1/git.1
  system: /Applications/Xcode.app/Contents/Developer/usr/share/man/man1/git.1
```

And the internal variable name `label` in the `--find` implementation is worth revisiting — `provider` is the meaningful term throughout all the documentation and design discussion, so the code should use it too unless there's a reason `label` is more general that I'm not seeing.

brew man openssl@3 openssl
```

List every location where `openssl(1)` is found (system, libressl, openssl@3):

```sh
brew man --list openssl
```

> List all man pages that the `libressl` formula provides:

Reviewer: Note the proposal above to make this behavior explicit with a new --all flag.
